options.go

```diff
@@ -23,3 +28,54 @@ func initConfig(option *WatcherOptions) {
 		option.Channel = "/casbin"
 	}
 }
+
+func MakeDefaultUpdateCallback(e *casbin.Enforcer) func(string) {
+	return func(msg string) {
+		msgStruct := &MSG{}
+
+		err := msgStruct.UnmarshalBinary([]byte(msg))
+		if err != nil {
+			log.Println(err)
+		}
+
+		e.EnableAutoNotifyWatcher(false)
```

Comment:
Why is it set to ``false``

Reply:
Or it will emit another update message 😵.
 A to B to A to B …

Reply:
It will be setback afterwards.

Reply:
Oops, I should respect the original auto-notify state… I'll fix it ASAP

Reply:
> Oops, I should respect the original auto-notification setting… I'll fix it ASAP

autoNotifyWatcher is a private attribute and there is no getter for it. It's impossible to do so.😑

Reply:
Just delete it, should not call EnableAutoNotifyWatcher

Reply:
> Just delete it, should not call EnableAutoNotifyWatcher

Let's suppose Alice adds a policy. Then Bob receives this notification and calls AddPolicy() to keep the consistence. But autoNotifyWatcher is set to be True by default and as a result, Bob will send another update notification to Alice. This is a dead end.

Reply:
You're right. But we need to add a new field instead of using ``autoNotifyWatcher`` because turn off ``autoNotifyWatcher`` in the callback will affect other parallel methods.

Reply:
You're right. But set ``EnableAutoNotifyWatcher`` false also affects other parallel methods. It needs a better way. I have no idea yet.

Reply:
> You're right. But set `EnableAutoNotifyWatcher` false also affects other parallel methods. It needs a better way. I have no idea yet.

How about depending on the duplication-avoided behavior? 

![image](https://user-images.githubusercontent.com/20621784/169248913-f9e89c4a-ccc5-4f86-9080-97cb089c5357.png)

Let's come back to Alice and Bob. The process will become `Alice -> Bob -> Alice` and then end due to duplication.

However removal-related APIs lack such checking. It requires a patch for Casbin Core to call `HasPolicies()` or use `RemovePoliciesWithEffected()` instead.
![image](https://user-images.githubusercontent.com/20621784/169250191-fc95f89b-f1ee-4fb4-b7f8-6064d5c68e71.png)

Reply:
That doesn't solve the problem. Errors may still occur if you AddPolicy and then RemovePolicy. What's more, how to solve UpdatePolicy? This behavior of calling twice is fundamentally incorrect.